internal/backcompatasset/asset.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2025 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backcompatasset
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+)
+
+// BackCompatAsset represents a wrapper around a backcompat asset executable.
+type BackCompatAsset struct {
```

Comment:
Can you follow pattern outlined/suggested in PR at https://github.com/palantir/godel-conjure-plugin/pull/636 instead?

Main difference is to define an interface for `BackCompatChecker` (like https://github.com/palantir/godel-conjure-plugin/pull/636/files), then have asset-based implementation be an unexported struct (https://github.com/palantir/godel-conjure-plugin/pull/636/files#diff-18e6ae998aa0e9b31c388c7e998467a0d9ce8bb7c34f73ef66a74676fc302af8R19) and have a function that creates it (https://github.com/palantir/godel-conjure-plugin/pull/636/files#diff-18e6ae998aa0e9b31c388c7e998467a0d9ce8bb7c34f73ef66a74676fc302af8R13) (the last one is analogous to the `New` function defined in this PR/file).